### ISSUE TITLE
nu.pattern.OpenCV.SharedLoader.SharedLoader() diagnoses Exception by Message - Fails on OpenJ9 - SOLUTION

### ISSUE THREAD

**Issue by markmaker** [openpnp/opencv#45]
Hi

As discussed here:
https://groups.google.com/d/msg/openpnp/p2xiolO3pf0/XeOrKAXfBQAJ
we cannot use OpenJDK+OpenJ9 for OpenPNP due to some class loader issue with OpenCV.

I found the reason. 

`nu.pattern.OpenCV.SharedLoader.SharedLoader()` diagnoses the exception by looking at the message. The message is slightly different with OpenJ9, therefore the exception is not correctly recognized and therefore falsely propagated:

https://github.com/openpnp/opencv/blob/fed49ee6be4726a8b4bca50153d8016f6b22330d/src/main/java/nu/pattern/OpenCV.java#L190-L201

The OpenJ9 VM generates this message:
`opencv_java342 (Not found in java.library.path)`

Suggestion: 

```
/* Only update the library path and load if the original error indicates it's missing from the library path. */
        String errorFragment = String.format("no %s in java.library.path", );
        if (ule == null || !(ule.getMessage().contains(Core.NATIVE_LIBRARY_NAME) 
              && ule.getMessage().contains("java.library.path"))) {
          logger.log(Level.FINEST, String.format("Encountered unexpected loading error."), ule);
          throw ule;
}
```

I don't know how to bypass maven dependencies locally therefore I made no PR.

_Mark

**Comment by markmaker** [comment]
Workaround (until this Issue is fixed):

1. Run OpenPNP with OpenJDK8+Hotspot.
2. While OpenPNP is still running, search the temporary directory for the OpenCV shared library (e.g. `%TEMP%\opencv_openpnp*\nu\pattern\opencv\windows\x86_64\opencv_java342.dll` on Windows)
3. Copy the file into one of your PATH directories (I used HOMEDIR`/maven/bin`)
4. Close OpenPNP.
5. Change the JRE to OpenJDK8+OpenJ9 
6. Run OpenPNP again. 
7. Enjoy the fast loading time. :-)

_Mark

**Comment by vonnieda** [comment]
Fixed in #54
